pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -517,7 +521,11 @@ pub mod pallet {
 				}
 			};
 
-			T::InboundQueue::submit(domain_address, incoming_msg)
+			for msg in incoming_msg.unpack() {
+				T::InboundQueue::submit(domain_address.clone(), msg)?;
+			}
```

Comment:
Weight of this function must take into account the maximum batch size possible. Defined by the deserialization implemenation.

Reply:
This is really difficult to measure because, right now, it's not measured. It's using hardcoded weights. The real weight of this does not depend on how many bytes we read from the message, it mostly depends of what messages are in the batch that will dispatch certain actions. i.e., if this message is an increase it will call to foreign investments and would imply swaps having a weight increase.

I'm not sure what kind of solution we want to follow here.

Reply:
I've got one which multiplies the current weight used by the number of messages in the batch